tests/src/Kernel/EntityMemoryCacheTest.php

```diff
@@ -0,0 +1,245 @@
+  protected $entityIdCache;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'system',
+    'apigee_edge',
+    'key',
+  ];
+
+  /**
+   * Tests developer entity memory cache.
+   */
+  public function testDeveloperEntityMemoryCache() {
+    $this->entityCache = $this->container->get('apigee_edge.controller.cache.developer');
+    $this->entityIdCache = $this->container->get('apigee_edge.controller.cache.developer_ids');
+
+    // Generate developer entities with random data.
+    $developers = [];
+    for ($i = 0; $i < 3; $i++) {
+      $id = $this->getRandomUniqueId();
+      $developers[$id] = new Developer([
+        'developerId' => $id,
+        'email' => strtolower($this->randomMachineName()) . '@example.com',
+      ]);
+    }
+
+    $this->assertFullCaches($developers);
+
+    /** @var \Apigee\Edge\Api\Management\Entity\DeveloperInterface $developer */
+    foreach ($developers as $developer) {
+      // Load developer by email and by id.
+      $this->assertSame($developer, $this->entityCache->getEntity($developer->getEmail()));
+      $this->assertSame($developer, $this->entityCache->getEntity($developer->getDeveloperId()));
+      $this->assertContains($developer->getEmail(), $this->entityIdCache->getIds());
+
+      // Remove developer from cache.
+      $this->entityCache->removeEntities([$developer->getDeveloperId(), $this->getRandomGenerator()->string()]);
+      $this->assertNull($this->entityCache->getEntity($developer->getEmail()));
+      $this->assertNull($this->entityCache->getEntity($developer->getDeveloperId()));
+      $this->assertNotContains($developer->getEmail(), $this->entityIdCache->getIds());
+    }
+
+    $this->assertEmptyCaches();
+  }
+
+  /**
+   * Tests app entity memory cache.
+   */
+  public function testAppEntityMemoryCache() {
+    $this->entityCache = $this->container->get('apigee_edge.controller.cache.apps');
+    $this->entityIdCache = $this->container->get('apigee_edge.controller.cache.app_ids');
+
+    // Generate developer app entities with random data.
+    $developer_apps = [];
+    $parent_id = $this->getRandomUniqueId();
+    for ($i = 0; $i < 2; $i++) {
+      $id = $this->getRandomUniqueId();
+      $developer_apps[$id] = new DeveloperApp([
+        'appId' => $id,
+        'name' => $this->getRandomUniqueId(),
+        'developerId' => $parent_id,
+      ]);
+    }
+
+    // Generate company app entities with random data.
+    $company_apps = [];
+    $parent_id = $this->getRandomUniqueId();
+    for ($i = 0; $i < 2; $i++) {
+      $id = $this->getRandomUniqueId();
+      $company_apps[$id] = new CompanyApp([
+        'appId' => $id,
+        'name' => $this->getRandomUniqueId(),
+        'companyName' => $parent_id,
+      ]);
+    }
+
+    $apps = $developer_apps + $company_apps;
+
+    $this->assertFullCaches($apps);
+
+    /** @var \Apigee\Edge\Api\Management\Entity\AppInterface $app */
+    foreach ($apps as $app) {
+      // Load app by id and by owner.
```

Comment:
Please extend the sentence because it is important what could be an owner.
... owner (developer uuid or company name).